### PR TITLE
[LOOP-440] Introduce BasalScheduleEditor

### PR DESCRIPTION
**Pull request by mpangburn**
- Support asynchronous schedule saving in `ScheduleEditor` & `ConfigurationPage`
- Allow `GuardrailWarning` captions to be overridden for special cases

Known bug: the whole value '0' in the basal fractional picker is always red, regardless of the fractional value. Will fix in a follow-up.

See also: https://github.com/tidepool-org/Loop/pull/105

### CODE REVIEW

File: LoopKitUI/Views/BasalRateScheduleEditor.swift [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    var crossedThresholds: [SafetyClassification.Threshold]
+
+    var body: some View {
+        assert(!crossedThresholds.isEmpty)
+        return GuardrailWarning(
+            title: crossedThresholds.count == 1 ? singularWarningTitle(for: crossedThresholds.first!) : multipleWarningTitle,
+            thresholds: crossedThresholds,
+            customCaption: { crossedThresholds in
+                crossedThresholds.allSatisfy({ $0 == .minimum })
+                    ? Text("A setting of 0 U/hr means you will not receive basal insulin or automation.", comment: "Warning text for basal rate of 0 U/hr")
```
Comment (mpangburn) [inline LoopKitUI/Views/BasalRateScheduleEditor.swift:110]: (Grabbed this copy from the Jira ticket, but I don't think it's accurate. You can still be high-temped with a basal setting of 0 U/hr, so you're still "receiving automation". Discussing now, will likely be updated.)

File: LoopKit/QuantityFormatter.swift [hunk 2]
```diff
@@ -147,7 +147,9 @@ public extension HKUnit {
     }
 
     var maxFractionDigits: Int {
-        if self == HKUnit.gram().unitDivided(by: .internationalUnit()) {
+        if self == .internationalUnitsPerHour {
+            return 3
```
Comment (nhamming) [inline LoopKit/QuantityFormatter.swift:151]: [nit] How does specific ranges of rates get affected by this? I mean, say between 5-10 IU/hr the pump can only take 0.01 resolution.

File: LoopKitUI/Views/ActivityIndicator.swift [hunk 3]
```diff
@@ -12,14 +12,18 @@ public struct ActivityIndicator: UIViewRepresentable {
 
     @Binding var isAnimating: Bool
     let style: UIActivityIndicatorView.Style
+    let color: UIColor?
     
-    public init(isAnimating: Binding<Bool>, style: UIActivityIndicatorView.Style) {
+    public init(isAnimating: Binding<Bool>, style: UIActivityIndicatorView.Style, color: UIColor? = nil) {
```
Comment (nhamming) [inline LoopKitUI/Views/ActivityIndicator.swift:17]: [nit] should this default to `nil` or the default activity indicator color?

File: LoopKitUI/Views/BasalRateScheduleEditor.swift (hunk 1 above)
Comment (nhamming) [inline LoopKitUI/Views/BasalRateScheduleEditor.swift:110]: [nit] does `.minimum` always mean `0 U/hr`?

**Review by nhamming — APPROVED** [review]
LGTM, just some questions in areas that I didn't fully understand.

File: LoopKitUI/Views/ActivityIndicator.swift (hunk 3 above)
Comment (mpangburn) [reply LoopKitUI/Views/ActivityIndicator.swift:17]: The `style` property implicitly sets `color`.

File: LoopKit/QuantityFormatter.swift (hunk 2 above)
Comment (mpangburn) [reply LoopKit/QuantityFormatter.swift:151]: This property determines the upper bound on digits; 3 won't be used if only 2 are necessary.

File: LoopKitUI/Views/BasalRateScheduleEditor.swift (hunk 1 above)
Comment (mpangburn) [reply LoopKitUI/Views/BasalRateScheduleEditor.swift:110]: In practice, _probably_, but I'll make this explicit.

Comment (mpangburn) [reply LoopKitUI/Views/BasalRateScheduleEditor.swift:110]: (Will make the change in my follow-up)

**Pull request merged**